dev/devicelab/lib/tasks/analysis.dart

```diff
@@ -5,112 +5,99 @@
-      file(path.join(flutterDirectory.path, 'analysis_benchmark.json'));
+  final bool watch;
 
-  @override
-  TaskResult get lastResult => new TaskResult.successFromFile(benchmarkFile);
-
-  @override
-  Future<num> run() async {
-    rm(benchmarkFile);
+  Future<double> call() async {
+    section('Analyze Flutter repo ${watch ? 'with watcher' : ''}');
+    final Stopwatch stopwatch = new Stopwatch();
     await inDirectory(flutterDirectory, () async {
-      await flutter('analyze', options: <String>[
+      final List<String> options = <String>[
         '--flutter-repo',
         '--benchmark',
-      ]);
+      ];
+
+      if (watch)
+        options.add('--watch');
+
+      stopwatch.start();
+      await flutter('analyze', options: options);
+      stopwatch.stop();
     });
-    return addBuildInfo(benchmarkFile,
-        timestamp: timestamp, expected: 25.0, sdk: sdk, commit: commit);
+    return stopwatch.elapsedMilliseconds / 1000;
   }
 }
 
-class FlutterAnalyzeAppBenchmark extends Benchmark {
-  FlutterAnalyzeAppBenchmark(this.sdk, this.commit, this.timestamp)
-      : super('analysis server mega_gallery');
+/// Times how long it takes to analyze the generated "mega_gallery" app.
+class _MegaGalleryBenchmark {
+  _MegaGalleryBenchmark({ this.watch = false });
 
-  final String sdk;
-  final String commit;
-  final DateTime timestamp;
+  final bool watch;
 
-  @override
-  TaskResult get lastResult => new TaskResult.successFromFile(benchmarkFile);
+  Future<double> call() async {
+    section('Analyze mega gallery ${watch ? 'with watcher' : ''}');
+    final Stopwatch stopwatch = new Stopwatch();
+    await inDirectory(_megaGalleryDirectory, () async {
+      final List<String> options = <String>[
+        '--benchmark',
+      ];
 
-  Directory get megaDir => dir(
-      path.join(flutterDirectory.path, 'dev/benchmarks/mega_gallery'));
-  File get benchmarkFile =>
-      file(path.join(megaDir.path, 'analysis_benchmark.json'));
+      if (watch)
+        options.add('--watch');
 
-  @override
-  Future<Null> init() {
-    return inDirectory(flutterDirectory, () async {
-      await dart(<String>['dev/tools/mega_gallery.dart']);
+      stopwatch.start();
+      await flutter('analyze', options: options);
+      stopwatch.stop();
     });
+    return stopwatch.elapsedMilliseconds / 1000;
   }
+}
 
-  @override
-  Future<num> run() async {
-    rm(benchmarkFile);
-    await inDirectory(megaDir, () async {
-      await flutter('analyze', options: <String>[
-        '--watch',
-        '--benchmark',
-      ]);
-    });
-    return addBuildInfo(benchmarkFile,
-        timestamp: timestamp, expected: 10.0, sdk: sdk, commit: commit);
+/// Runs a [benchmark] several times and reports the best result.
+Future<double> _run(_Benchmark benchmark) async {
+  double minValue;
+  for (int i = 0; i < _kRunsPerBenchmark; i++) {
+    // Delete cached analysis results.
+    rmTree(dir('${Platform.environment['HOME']}/.dartServer'));
```

Comment:
I think we have to do this before each run of the analyzer, not before each group of runs.